pyxis_slurmstepd.c

```diff
@@ -251,6 +264,85 @@ static int enroot_create_user_runtime_dir(void)
 	return (0);
 }
 
+/* As root, create the per-user directory where squashfs files are stored and later could be reused. */
+static int enroot_create_image_save_dir(char *path) 
+{
+	int ret;
+	char uid_dir_path[PATH_MAX];
+
+	if (path == NULL || strlen(path) == 0) {
+		slurm_error("pyxis: path is empty");
+		return (-1);
+	}
+
+	/* In case it's a dir, we create per user folder */
+	if (path[strlen(path) - 1] == '/') {
```

Comment:
can path have empty spaces at the end? if so, should we trim them?

Reply:
added validation in `srun`